airbyte-webapp/src/views/Connector/ConnectorForm/useAuthentication.tsx

```diff
@@ -140,7 +143,27 @@ export const useAuthentication = (): AuthenticationHook => {
 
   const spec = connectorSpec?.connectionSpecification as JSONSchema7;
 
-  const valuesWithDefaults = useMemo(() => getValues(values), [getValues, values]);
+  const getValuesSafe = useCallback(
+    (values: ConnectorFormValues<unknown>) => {
+      try {
+        // We still see cases where calling `getValues` which will eventually use the yupSchema.cast method
+        // crashes based on the passed in values. To temporarily make sure we're not crashing the form, we're
+        // falling back to `values` in case the cast fails. This is a temporary patch, and we need to investigate
+        // all the failures happening here properly.
```

Comment:
> To temporarily make sure we're not crashing the form, we're falling back to `values` in case the cast fails.

What are the consequences of falling back to `values` in this case? Does this just mean that default values will not be inserted, or are there any others?

Reply:
afaik it should only cause the default values to not be present, and just have the Formik values "as it". Ideally we still figure out quick enough what cases might fail here (since the weird S3 spec might not the only one) and try to fix them properly, but that way we have at least a chance it's still working "largely" (given that even without the default value, we saw most OAuth connectors working).